src/internalBinding/timers.cc

```diff
@@ -23,17 +23,18 @@ static bool enqueueWithDelay(JSContext *cx, unsigned argc, JS::Value *vp) {
   JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
   JS::HandleValue jobArgVal = args.get(0);
   double delaySeconds = args.get(1).toNumber();
+  bool repeat = args.get(2).toBoolean();
 
   // Convert to a Python function
   JS::RootedValue jobArg(cx, jobArgVal);
   PyObject *job = pyTypeFactory(cx, jobArg)->getPyObject();
   // Schedule job to the running Python event-loop
   PyEventLoop loop = PyEventLoop::getRunningLoop();
   if (!loop.initialized()) return false;
-  PyEventLoop::AsyncHandle::id_ptr_pair handler = loop.enqueueWithDelay(job, delaySeconds);
+  PyEventLoop::AsyncHandle::id_t handleId = loop.enqueueWithDelay(job, delaySeconds, repeat);
 
   // Return the `timeoutID` to use in `clearTimeout`
```

Comment:
```suggestion
  Py_DECREF(job):
  // Return the `timeoutID` to use in `clearTimeout`
```

Reply:
The reference count shouldn't be decreased here as the job function will be called later.

Reply:
a new reference to job is created in `enqueueWithDelay` by `PyCFunctionNew`

Reply:
just checked, and yes the refcount is 1 when `job` is defined, and 2 after `enqueueWithDelay` is called

Reply:
@caleb-distributive You're right. Confirmed by gdb.